parlai/tasks/wizard_of_wikipedia/agents.py

```diff
@@ -757,6 +758,107 @@ def get(self, episode_idx, entry_idx=0):
         return a
 
 
+class WikiPageTitleTeacher(WizardDialogKnowledgeTeacher):
+    """
+    Generates the title of Wikipedia page used as source of knowledge.
+
+    The context provided by this teacher (`text`) is the conversation history, with chosen topic removed.
+    The label is the title of the Wikipedia page of the passage that wizard selected for crafting
+    the next utterance; in other words, the source of knowledge for this utterance.
+    """
+
+    def __init__(self, opt, shared=None):
+        self.opt = copy.deepcopy(opt)
+        self.opt['label_type'] = 'response'
+        super().__init__(self.opt, shared=shared)
+        self.id = 'WikiTitleGenerationTeacher'
+        self._conv_history_len = self.opt['conversation_history_length']
+        self._skip_no_title = self.opt['skip_no_title']
+        if not shared:
+            self._preprocess_data()
+        else:
+            self.titles_data = shared['titles_data']
+
+    @classmethod
+    def add_cmdline_args(cls, parser, partial_opt=None):
+        super().add_cmdline_args(parser, partial_opt=partial_opt)
+        agent = parser.add_argument_group('Wikipedia Page Title Arguments')
+        agent.add_argument(
+            '--conversation-history-length',
+            type=int,
+            default=0,
+            help='Number of previous utterances to keep in context, 0 (default) includes all',
```

Comment:
can we make the default -1? and have that map to including all? making 0 mean includes all doesn't seem right (as one could argue this should include... none)

Reply:
I changed default to `-1`, and added something to change any non-positive value to that.